### DISCUSSION TITLE
Improvements in Generated Audio Quality According to the Implementation Method in xtts v2

### DISCUSSION THREAD

**Discussion by dancodedev** [coqui-ai/TTS#3209] [General Q&A]
Recently, I had the opportunity to experiment with version 1 of the xtts TTS model. During my tests, I noticed a significant improvement in voice synthesis quality when I used the model directly (Method 3), compared to more simplified methods such as the TTS API (Method 1) or the command line interface (CLI, Method 2).

This observation leads me to ask if it is reasonable to expect a similar difference in quality when using the xtts v2 model. Is it possible that direct access to the model and customization of settings offers superior quality in voice synthesis?

Additionally, I am particularly interested in knowing what parameters can be adjusted in the CLI and API formats to improve the quality of the synthesized voice, without worrying about processing time. Are there specific options or settings in these methods that allow reaching a quality level comparable to that obtained through direct initialization and use of the model?

I appreciate any guidance or tips on how to maximize the quality of the voice generated using the xtts v2 model, especially regarding parameter adjustments in the different implementation methods.

## **Method 1: Using the TTS API**

```python
from TTS.api import TTS
tts = TTS("tts_models/multilingual/multi-dataset/xtts_v2", gpu=True)

tts.tts_to_file(
    text="It took me quite a long time to develop a voice, and now that I have it I'm not going to be silent.",
    file_path="output.wav",
    speaker_wav="/path/to/target/speaker.wav",
    language="en"
)
```

This method offers a high-level interface for working with the `xtts` model, ideal for those who prefer quick and simple integration into their applications.

## **Method 2: Command Line Interface (CLI)**

```bash
tts --model_name tts_models/multilingual/multi-dataset/xtts_v2 \
    --text "Bugün okula gitmek istemiyorum." \
    --speaker_wav /path/to/target/speaker.wav \
    --language_idx tr \
    --use_cuda true
```

The CLI interface is useful for quick tests or for users who prefer to work directly from the command line, offering flexibility and ease of use.

## **Method 3: Direct Initialization and Use of the Model**

```python
from TTS.tts.configs.xtts_config import XttsConfig
from TTS.tts.models.xtts import Xtts

config = XttsConfig()
config.load_json("/path/to/xtts/config.json")
model = Xtts.init_from_config(config)
model.load_checkpoint(config, checkpoint_dir="/path/to/xtts/", eval=True)
model.cuda()

outputs = model.synthesize(
    "It took me quite a long time to develop a voice and now that I have it I am not going to be silent.",
    config,
    speaker_wav="/data/TTS-public/_refclips/3.wav",
    gpt_cond_len=3,
    language="en",
)
```

This approach provides the most control over the model, allowing fine adjustments and advanced customizations. It is especially useful for users with technical knowledge who seek to experiment or optimize voice synthesis quality.

**Comment by FurkanGozukara** [comment]
How did you manage to improve quality over other 2 methods?

How it is possible?

**Reply by dancodedev** [reply]
I have used the tool available at https://huggingface.co/spaces/coqui/xtts, where the model is directly employed. In my experience, the quality in the xtts v1.1 version was significantly superior compared to the tests conducted using the command-line interface. Although in version 2.0 this difference is not as noticeable, I'm curious to know if anyone else has observed this variation in quality

**Reply by FurkanGozukara** [reply]
I see

currently I am doing fine tuning what you think about quality?


https://github.com/coqui-ai/TTS/assets/19240467/95efe0d7-f77a-456b-be0e-2533d9138fb7

**Reply by dancodedev** [reply]
Please excuse me, but my tests are in Spanish, and it is in this language that I can truly assess whether the quality is good or bad. It's possible that these differences do not exist in English, which is why there might not be more people who have experienced this. Therefore, I cannot determine whether the example you're showing me is good or bad, as my knowledge of English is quite limited

**Comment by ukemamaster** [comment]
@dancodedev Have you tried to fine tune the xtts-v2 model with some spanish data? I am trying to do so, and i get some breathing noises and sudden increase in the volume of generated audios. What is your experience on that?

**Reply by dancodedev** [reply]
Hello, with version v2.02 the Spanish sounds extremely good but if you use cli it fails a lot you have to use the "API" mode inference it is more stable
